geomstats/geometry/pullback_metric.py

```diff
@@ -111,6 +112,47 @@ def pickable_inner_product(i, j):
         metric_mat = gs.reshape(gs.array(out), (-1, self.dim, self.dim))
         return metric_mat[0] if base_point.ndim == 1 else metric_mat
 
+    def inner_product_derivative_matrix(self, base_point=None):
+        r"""Compute the inner-product derivative matrix.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., *shape]
+            Base point.
+            Optional, default: None.
+
+        Returns
+        -------
+        inner_prod_deriv_mat : array-like, shape=[..., dim, dim, dim]
+            Inner-product derivative matrix.
+        """
+        hessian_aij = []
+        jacobian_ai = []
```

Comment:
Since we need the jacobian and the hessian, I think we should add a function to the backends called e.g. `jacobian_and_hessian` (same idea as e.g. `value_and_grad`), because the hessian needs to compute the jacobian and when we need both, we are basically repeating a computation that we have already done (we are committing this "repeating" error in `value_and_grad`... I'm preparing a PR to fix it).

For `autograd` it should be simple, because the hessian is simply implemented as: `jacobian(jacobian(fun))(x)`.

For `tensorflow` it is similar, because we are the ones computing the jacobians.

For `torch` it may require a little more thought, as they have additional arguments to (supposedly) run things faster, but it should not be very difficult (they're code is relatively straighforward).

What do you think?

Reply:
Good point. I've implemented it for autograd and tensorflow (and numpy, returning the error message).

I left a placeholder for torch, which basically calls jacobian and hessian separately. I'll leave that one for future optimization if that is fine with you. I will raise an issue.